File: Algo_Method/calc_bits/ch10/Q3/Q3.py
n, m = gets.chomp.split.map(&:to_i)
w = gets.chomp.split.map(&:to_i)
v = list(map(int, input().split()))
max_value = 0
for bits in range(1<<n):
    sum_w, sum_v = 0, 0
    for i in range(n):
        if bits & (1<<i): 
            sum_w += w[i]
            sum_v += v[i]
    if sum_w > m: continue
    max_value = max(max_value, sum_v)
print(max_value)
    

File: Algo_Method/calc_bits/ch10/Q3/Q3.rb
n, m = gets.chomp.split.map(&:to_i)
w = gets.chomp.split.map(&:to_i)
v = gets.chomp.split.map(&:to_i)
max_value = 0
(1<<n).times do |bits|
    sum_w = 0
    sum_v = 0
    n.times do |i|
        if bits & (1<<i) != 0
            sum_w += w[i]
            sum_v += v[i]
        end
    end
    next if sum_w > m
    max_value = [max_value, sum_v].max
end
p max_value